include/motor-control/core/tasks/motion_controller_task.hpp

```diff
@@ -187,8 +212,22 @@ class MotionControllerTask {
 
     [[nodiscard]] auto get_queue() const -> QueueType& { return queue; }
 
+    // also create top level query msg
+    void run_diag0_interrupt() {
+        if (!driver_error_handled()) {
+            static_cast<void>(
+                queue.try_write_isr(can::messages::MotorDriverErrorEncountered{
+                    .message_index = 0}));
+        }
+    }
+
+    auto driver_error_handled() -> bool {
+        return driver_error_handled_flag.exchange(true);
```

Comment:
what resets this flag? as coded, it looks like a given microcontroller will only ever respond to a driver error once until it's rebooted. we probably need a way to reset this when the error goes away.